docs/faq/unique-element-ids.md

```diff
@@ -30,3 +30,34 @@ const id = newId()
 See it on the [SFC Playground](https://play.vuejs.org/#eNqVU99P2zAQ/ldOFlJhpQ0T2ksWKm2MSeyBIeAxD2TxpTU4tmU7bVGU/31nuy1Fgk6LFMm+++7X95179s2Y6bJDlrPC1VYYDw59Z2alEq3R1sMDrv1PgZJDY3ULo2m2s4TAUamKLEVSDF08tkZWHukGUIREQN+lFPUzaAUVyOoPSvAaGl13DvwCQSjT+YjPUkDBxRJqWTl3UbKVrUzJNomK135iIvLfVC2WDLKPEXda/gPxA9MMQqsdsMioCzoV2d5MdHX+RSK4WhvkZJmG/qAPEVw4wr3kNJAUCieNxPXX4AiHCRcW61Ahh1rLrlXkGkLGlGEGn2BMf8zUVnYu1MRrk8MXE5IQkogOpWfslL3R4GPxelC4uuYw7MQTfPrkSLRScWyox1urjTuONSMXOdx7K9ScCp4EVK2V8yA4XKRUx2Q9JHiiLLKchM4bbYlgwUnCvt+oPwxFFk9bbNwAyAXfQv9DgLArb+nf8B7JTQWjP5hz+BwVMRXnNObEivnCk/HsHZIjV0SuRE+KdcoTCWchJ64ju02nop4bZowlRteEGQk+GZ2kmpYEsQoej/rkHo768TgmGx6pHlXxNIhqxJxqaUXVYljJat0aIdH+jkvpSpanhMFXSalXv6LN2w5Pt/Z6gfXzO/Yntw62kt1adGiX9Bh2Pk+bhj65r+5vaK/2nK3mXXg6B5x36GiZ08MJsO+d4tT2Hi52ex0Xkhh/cFdrj8pthwqNBuQQ8SWjfb48MPpru+fT8xgXWBz+AnDCnEI=).
 
 An element `id` beginning with a digit [can cause problems](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id), so we put a prefix before the count.
+
+## SSR limitations
+
+One potential problem with the above approach is that if your app is [server-side rendered](https://vuejs.org/guide/scaling-up/ssr.html), and *especially* if you have async components, the `count` may not be identical between the server and the client if the calls to `newId` happen in a different order. This could lead to mismatches among references to the same IDs (because the "same" id evaluated on the server might be different since, to enhance performance, Vue's hydration diffing doesn't take attribute values into account when patching differences between server and client). This can be solved by recreating the `id` (or only creating it) on the _client_ once the component mounts. There is actually an [example of this in the Vue docs for SSR](https://vuejs.org/guide/scaling-up/ssr.html#custom-directives). We could combine it with the approach above:
+
+```vue
+<template>
+<label :for="id">
+<input v-id="id">
+</template>
```

Comment:
This is not a valid template.
A label needs to be closed at the very least and you probably need a root element as well.

```suggestion
<template>
<div>
  <label :for="id">Label</label>
  <input v-id="id" />
</div>
</template>
```